### PR TITLE
implement set, test, star, subseteq, and add README. add tests for star

### PR DESCRIPTION
**Pull request by etw33**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Overview
+This package implements hash-consed binary decision diagrams (BDDs) and identity-suppressed decision diagrams (IDDs).
+
+An IDD, like a BBD, can be seen as representing a transition relation R on a state space of boolean vectors. I.e. boolean vector pair (v1, v2) belongs to R if and only if evaluating the IDD-representation of R in the environment given by (v1, v2) yields true.
+
+The main motivation for IDDs is that they represent the identity relation in a constant amount of space instead of in an amount of space that is linear in the size of the elements of the state space.
```
Comment (smolkaj) [inline README.md]: Linear in the number of variables, exponential in the size of the state space.

File: src/idd.ml [hunk 2]
```diff
@@ -87,6 +87,13 @@ let branch (mgr : manager) (x : Var.t) (hi : t) (lo : t) : t =
     end
   (* ) *)
 
+let test mgr i b =
+    branch mgr (Var.inp i) (if b then ident else empty)
```
Comment (smolkaj) [inline src/idd.ml]: Maybe build the tree straight from the constructors here?
```
Dd.branch mgr.dd x
  (if b then Dd.True else Dd.False)
  (if b then Dd.False else Dd.True)
```

File: src/idd.ml [hunk 3]
```diff
@@ -87,6 +87,13 @@ let branch (mgr : manager) (x : Var.t) (hi : t) (lo : t) : t =
...
+          (if b then empty else ident)
+
+let set mgr i b =
```
Comment (smolkaj) [inline src/idd.ml]: same as above

File: src/idd.ml [hunk 4]
```diff
@@ -173,6 +180,16 @@ let rec seq mgr (d0:t) (d1:t) =
...
 
+let star mgr (d0:t) =
+  let rec loop curr prev =
+    if equal curr prev then prev else
+      loop (seq mgr curr curr) curr
+  in
+  loop (union mgr ident d0) ident
+
+let subseteq mgr (d0:t) (d1:t) =
+    equal (union mgr d0 d1) d1
```
Comment (smolkaj) [inline src/idd.ml]: nit: indent by two spaces

**Review by smolkaj — COMMENTED** [review]
Looks great! Just some minor suggestions.

File: README.md (hunk 1 above)
Comment (etw33) [reply README.md]: I replaced "elements of the state space" with  "boolean vectors" aka number of variables

**Comment by smolkaj** [comment]
LGTM!

**Pull request merged**